patches/api/0419-Paper-Item-API.patch

```diff
@@ -0,0 +1,446 @@
++
++    // BundleMeta
++    PropertyKey<List<ItemStack>> ITEMS = PropertyKey.of("Items", TypeFactory.parameterizedClass(List.class, ItemStack.class));
++
++    // BookMeta
++    PropertyKey<String> TITLE = PropertyKey.of("title", String.class);
++    PropertyKey<String> TITLE_FILTERED = PropertyKey.of("filtered_title", String.class);
++    PropertyKey<String> AUTHOR = PropertyKey.of("author", String.class);
++    PropertyKey<List<Component>> PAGES = PropertyKey.of("pages", TypeFactory.parameterizedClass(List.class, Component.class));
++    PropertyKey<List<Component>> PAGES_FILTERED = PropertyKey.of("filtered_pages", TypeFactory.parameterizedClass(List.class, Component.class));
++    PropertyKey<Integer> GENERATION = PropertyKey.of("generation", Integer.class);
++    PropertyKey<Boolean> RESOLVED = PropertyKey.of("resolved", Boolean.class);
++
++    // UnsignedBookMeta
++    PropertyKey<List<String>> PAGES_WRITTEN = PropertyKey.of("pages", TypeFactory.parameterizedClass(List.class, String.class));
++
++    // BlockStateMeta
++    PropertyKey<PropertyHolder.Mutable> BLOCK_ENTITY_TAG = PropertyKey.of("BlockEntityTag", PropertyHolder.Mutable.class);
++    // Some odd entity tags included in item meta
++    PropertyKey<NamespacedKey> NOTE_BLOCK_SOUND = PropertyKey.of("note_block_sound", NamespacedKey.class);
++
++    // AxolotlBucket
++    PropertyKey<Axolotl.Variant> VARIANT = PropertyKey.of("Variant", Axolotl.Variant.class);
++
++    // MusicInstrumentMeta
++    PropertyKey<MusicInstrument> INSTRUMENT = PropertyKey.of("instrument", MusicInstrument.class);
++
++}
+diff --git a/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java b/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..4861b9e3c431e3e9d61aa3949fed61c68869357d
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/item/properties/StewEffect.java
+@@ -0,0 +1,8 @@
++package io.papermc.paper.inventory.item.properties;
++
++import org.bukkit.potion.PotionEffectType;
++import org.jetbrains.annotations.ApiStatus;
++
++@ApiStatus.Experimental
++public record StewEffect(PotionEffectType effect, int duration) {
++}
+diff --git a/src/main/java/io/papermc/paper/inventory/item/properties/TropicalFishVariant.java b/src/main/java/io/papermc/paper/inventory/item/properties/TropicalFishVariant.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..061e2627e2f1202504a214edb2f5764a8a757298
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/item/properties/TropicalFishVariant.java
+@@ -0,0 +1,9 @@
++package io.papermc.paper.inventory.item.properties;
++
++import org.bukkit.DyeColor;
++import org.bukkit.entity.TropicalFish;
++import org.jetbrains.annotations.ApiStatus;
++
++@ApiStatus.Experimental
++public record TropicalFishVariant(DyeColor patternColor, DyeColor bodyColor, TropicalFish.Pattern type) {
```

Comment:
I think we settled on not exposing records directly to the API. This should be an interface with a package-private record impl (same of all of these). Using interfaces improves documentation and allows for easier, better changes without breaking API.